validity/src/db/types.rs

```diff
@@ -127,6 +128,11 @@ impl OPSuccinctRequest {
             .get_l2_block_data_range(start_block as u64, end_block as u64)
             .await?;
 
+        let l1_head_block_number = fetcher
+            .get_l1_head_with_safe_head(end_block as u64)
+            .await?
+            .1;
```

Comment:
If we're going to do it this way, then we should update the places where we get the host args to ensure that this actually matches.

Currently, we fetch the `l1_head` on the fly.